Lib/asyncio/sslproto.py

```diff
@@ -460,7 +460,8 @@ def eof_received(self):
                 logger.debug("%r received EOF", self)
 
             if self._state == SSLProtocolState.DO_HANDSHAKE:
-                self._on_handshake_complete(ConnectionResetError)
+                self._on_handshake_complete(
+                    ConnectionResetError("SSL Connection reset by peer"))
```

Comment:
I'm not convinced that it's always the case. SSLProtocol._do_shutdown() calls _call_eof_received().

If you want to pass a reason, I suggest adding a parameter to _call_eof_received(). And maybe add a private _eof_received() function which accepts a message. For example, eof_received() can simply call _eof_received() with a default message.

Reply:
I don't think we have other reasons with the connection is being reset here.
`ConnectionResetError` here is only raised if the state of the connection is `SSLProtocolState.DO_HANDSHAKE`.

The `_call_eof_received()` method can be called only by `_do_shutdown()` and `_do_read()`
* `_do_shutdown()` is only called if the state is `SSLProtocolState.SHUTDOWN`.
* `_do_read()` is only called if the state is `SSLProtocolState.WRAPPED`.

So this exception can only be raised at the early stage of the SSL connection establishment.